src/NuGet.Clients/PackageManagement.VisualStudio/SolutionRestore/SolutionRestoreJob.cs

```diff
@@ -179,21 +190,32 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob, IDisposable
                 var solutionDirectory = _solutionManager.SolutionDirectory;
                 var isSolutionAvailable = _solutionManager.IsSolutionAvailable;
 
-                // make sure all projects are loaded before start to restore. Since
-                // projects might not be loaded when DPL is enabled.
-                _solutionManager.EnsureSolutionIsLoaded();
+                // Check if solution has deferred projects
+                var packageReferencesDict = new Dictionary<PackageReference, List<string>>(new PackageReferenceComparer());
```

Comment:
Better use case will be to check for IsSolutionDPLEnabled inside SolutionHasDeferredProjects api. Will update accordingly.